### ISSUE TITLE
Ability to customize Clock

### ISSUE THREAD

**Issue by eRfO** [kagkarlsson/db-scheduler#201]
It would be possible to configure the `com.github.kagkarlsson.scheduler.Clock` with the `SchedulerBuilder ` and in the case of Spring Boot like a bean?

Thanks

**Comment by kagkarlsson** [comment]
It could be exposed I suppose, what is your use-case? It defaults to `new SystemClock()`

**Comment by eRfO** [comment]
To use a non UTC clock like `OffsetDateTime.now().toInstant()`. Does this make sense?

**Comment by kagkarlsson** [comment]
Not sure :), what are you hoping to achieve? `OffsetDateTime.now().toInstant()` is the same instant in time as `Instant.now()`.

**Comment by eRfO** [comment]
You're right, the instant is always in UTC. So I don't now, I had changed the clock because it seemed to be behaving the wrong behavior.

**Comment by asmolinski-allegro** [comment]
Hey, bumping the question :)
Our use-case is the following: we have time-dependent feature that becomes effective starting from a certain date. We want to have an integration test that will simulate execution on that particular date. In our app Clock is a Spring bean and we can easily change it to a fixed one. The problem happens when dbscheduler comes into play as it has hardcoded systemDefault. So if we mock clock e.g. 1 month in the future, the task will be scheduled in the future, but it won't be executed until that time. I tried solution with `scheduler.reschedule` but it's quite cumbersome as we have a flow where task can schedule another task(s), so I would have to reschedule the first task, wait for it to finish, then reschedule the next tasks that it had spawned.
Having the ability to override the clock would make integration testing of Spring apps that use dbscheduler much easier.

**Comment by kagkarlsson** [comment]
For testing, have you checked the possibility of using the `ManualScheduler` instead? (a bit undocumented, but using it in tests to be able to affect clock and run due when I want to)
Using a normal scheduler in tests will be a bit error prone since it is async and run in completely different threads. The `ManualScheduler.runAnyDueExecutions()` runs in the calling thread, so synchronously

See example
https://github.com/kagkarlsson/db-scheduler/blob/master/db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/functional/DisabledCronTaskTest.java#L57

**Comment by asmolinski-allegro** [comment]
Hi @kagkarlsson, thanks for your prompt response!
The ManualScheduler looks like an interesting option, I will investigate it.

**Comment by deximat** [comment]
Here is my proposed PullRequest: 
https://github.com/kagkarlsson/db-scheduler/pull/720

**Comment by kagkarlsson** [comment]
Closing as done
